src/components/Contacts/ContactDetails/ContactDonationsTab/PartnershipInfo/EditPartnershipInfoModal/EditPartnershipInfoModal.tsx

```diff
@@ -150,6 +150,10 @@ export const EditPartnershipInfoModal: React.FC<
           pledgeStartDate: attributes.pledgeStartDate?.toISODate() ?? null,
           nextAsk: attributes.nextAsk?.toISODate() ?? null,
           primaryPersonId: attributes.primaryPersonId,
+          pledgeAmount:
+            (attributes.pledgeAmount as any) === ''
+              ? null
+              : Number(attributes.pledgeAmount),
```

Comment:
What do you think about this to avoid the `any` cast and parsing a number that is already a number?

```suggestion
            typeof attributes.pledgeAmount === 'number'
              ? attributes.pledgeAmount
              // When the field is blank, the value will be an empty string, even though TypeScript says the type is `number | null | undefined`
              : null,
```

Reply:
I wonder if you could chnage it to the below as it would catch `empty strings`, `null` and `undefined` which are the types for `attributes.pledgeAmount`
```suggestion
            !attributes.pledgeAmount
              ? null
              : Number(attributes.pledgeAmount),
```